### ISSUE TITLE
Derived TypedDict should allow field type override if field type is `Any`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6590]
The following code should not produce an error, but currently does:

```python
from typing import Any, TypedDict

class A(TypedDict):
    x: Any

class B(A):
    x: int
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
